bots/mainbot/pilgrim.js

```diff
@@ -23,23 +24,41 @@ const dirs = [
     { x: 1, y: 1 },
     { x: -1, y: 1 },
 ];
+
+pilgrim.create_resource_map = (self) => {
+	let resource_map = new Array(self.map.length);
+	for (let i = 0; i < self.map.length; i++) {
+		resource_map[i] = new Array(self.map[0].length);
+	}
+
+	let i, j;
+  for (i = 0; i < self.map.length; i++) {
```

Comment:
Is it just me or is this missing indentation?

Reply:
Fixed. Mix of tabs and spaces for indents made weird things happen. Updated everything to use tabs.